types/document.d.ts

```diff
@@ -1,9 +1,41 @@
+/// <reference path="./utilities.d.ts" />
 declare module 'mongoose' {
   import mongodb = require('mongodb');
 
   /** A list of paths to skip. If set, Mongoose will validate every modified path that is not in this list. */
   type pathsToSkip = string[] | string;
 
+  /**
+   * @summary Lean document base type.
+   */
+  interface LeanDocumentBaseType {
+    _id: Types.ObjectId;
```

Comment:
This isn't sufficiently flexible. Not all lean documents have `_id` as an ObjectId. Maybe `interface LeanDocumentBaseType<IdType = Types.ObjectId> { _id: IdType; }` ?